docs/datamodel/links.rst

```diff
@@ -14,16 +14,265 @@ There are two kinds of link item declarations: *abstract links*, and
 The number of targets is specified by the keywords :ref:`required
 <ref_eql_ddl_links_syntax>`, :ref:`single <ref_eql_ddl_links_syntax>`,
 and :ref:`multi <ref_eql_ddl_links_syntax>`.  It is also possible to
 restrict how many source objects can link to the same target via the
-:eql:constraint:`exclusive` constraint.
+:eql:constraint:`exclusive` constraint. Using these tools it's
+possible to specify common relationships between things:
+*one-to-one*, *one-to-many*, and *many-to-many*.
+
+
+One-to-One
+----------
+
+A *one-to-one* relationship represents a situation where everything
+is broken up into pairs and all the pairs are unique. For example,
+``Employee`` and ``ReservedParking``:
+
+.. code-block:: sdl
+
+    type Employee {
+        required property name -> str;
+        single link parking -> ReservedParking {
+            constraint exclusive;
+        }
+    }
+    type ReservedParking {
+        required property number -> int64;
+    }
+
+An ``Employee`` can have up to one ``ReservedParking`` assigned
+exclusively to them. The :eql:constraint:`exclusive` constraint
+ensures that no more than *one* ``Employee`` can get the same
+``ReservedParking``, while the ``single`` qualifier on the link (which
+is the default, so it can be omitted) ensures that no ``Employee`` can
+have more than *one* ``ReservedParking``. Together the constraint and
+the qualifier specify a *one-to-one* relationship.
+
+Although the link is specified only on one of the objects, the
+relationship involves both of them and so it can be accessed from
+either end. To get the assigned ``ReservedParking`` given an
+``Employee`` the following query can be used:
+
+.. code-block:: edgeql
+
+    WITH Alice := (
+        SELECT Employee FILTER .name = 'Alice'
+    )
+    SELECT Alice.parking {
+        number
+    };
+
+The reverse lookup of who owns a particular ``ReservedParking`` spot
+can be done by using a backward link traversal like so:
+
+.. code-block:: edgeql
+
+    WITH Spot := (
+        SELECT ReservedParking FILTER .number = 42
+    )
+    SELECT Spot.<parking[IS Employee] {
+        name
+    };
+
+Backward link traversal requires to specify the original link's source
+type, but other than that it works the same way as forward traversal.
+
+
+One-to-Many
+-----------
+
+A *one-to-many* relationship lacks the symmetry of a *one-to-one*,
+instead represents situations like ownership or hierarchies. For
+example, ``Person`` and ``Shoes``:
+
+.. code-block:: sdl
+
+    type Person {
+        required property name -> str {
+            constraint exclusive;
+        }
+        multi link cars -> Shoes {
+            constraint exclusive;
+        }
+    }
+    type Shoes {
+        required property description -> str {
+            # Just making sure that each description
+            # is unique like a name.
+            constraint exclusive;
+        }
+    }
+
+A ``Person`` can have potentially have more than one pair of
```

Comment:
I think `Shoes` is an inconvenient example in the context of the English language, because you use the term "pair of ``Shoes``" when talking about a single object, so this might be somewhat confusing to non-native speakers.

Reply:
Yeah. I'm looking for something that would be exclusively owned in a one-to-many way. So that's why I went for a personal item.

Maybe a ``Shirt`` or a ``Phone`` then? With ``Phone`` the example of getting multiple new ones may be a bit weird.